### PR TITLE
Timelapse mode

### PR DESCRIPTION
**Pull request by ladyada**
@jedgarpark this is a new timelapsing mode with sub-modes use OK to turn on / off timelapse, and SEL button to change power mode (backlight/preview)

### CODE REVIEW

**Review by jedgarpark — APPROVED** [review]
Works great!

**Review by jepler — APPROVED** [review]
Good to merge once the linter is happy. LMK if you want me to do that bit.

**Comment by ladyada** [comment]
@jedgarpark dunzo - please check latest commit code/library for 'frozen' settings

File: examples/camera/code.py [hunk 1]
```diff
@@ -34,6 +37,41 @@
...
         )
         pycam.blit(last_frame)
+    elif pycam.mode_text == "LAPS":
+        if timelapse_remaining is None:
+            pycam._timelapsestatus_label.text = "STOP"
+        else:
+            timelapse_remaining = timelapse_timestamp - time.time()
+            pycam._timelapsestatus_label.text = f"{timelapse_remaining}s /    "
+        pycam._timelapse_rate_label.text = pycam._timelapse_rate_label.text
+        pycam._timelapse_submode_label.text = pycam._timelapse_submode_label.text
```
Comment (jepler) [inline examples/camera/code.py:47]: Were these re-assignments of the label text really needed? I'll try testing without it.

Comment (ladyada) [reply examples/camera/code.py:47]: yes because otherwise display refresh doesnt pick up the labels after the screen has been blitted manually

Comment (jepler) [reply examples/camera/code.py:47]: Makes sense, will add a comment to that effect

**Comment by jepler** [comment]
There's some minor flickering on the screen when the new labels are in use but nothing that makes me want to hold this back

**Comment by jedgarpark** [comment]
@ladyada just got back from trip, tested timelapse, the new locked settings are great, no more flicker!

**Pull request merged**